pkg/streamingpromql/operators/functions/absent.go

```diff
@@ -95,14 +82,18 @@ func (a *Absent) NextSeries(_ context.Context) (types.InstantVectorSeriesData, e
 	output := types.InstantVectorSeriesData{}
 
 	var err error
-	output.Floats, err = types.FPointSlicePool.Get(a.timeRange.StepCount, a.memoryConsumptionTracker)
-	if err != nil {
-		return output, err
-	}
-
+	idx := 0
 	for step := range a.timeRange.StepCount {
 		t := a.timeRange.IndexTime(int64(step))
 		if !a.presence[step] {
+			// We only allocate the slice once we know we have point to add.
+			if idx == 0 {
```

Comment:
This can be simplified further: we can check if `output.Floats` is `nil`, and if it is, then that's our signal to get the slice from the pool. Then we don't need `idx` at all.

Reply:
Fixed in aa5df1c42dc1daaf8ff1d2245149e64800c322b6